### PR TITLE
Extract features tidy

### PR DESCRIPTION
**Pull request by jonthegeek**
Update `extract_features` to return two tibbles, 1 for outputs, 1 for attention.

### CODE REVIEW

**Comment by jonthegeek** [comment]
I've integrated the functions from RBERTviz for the layer outputs. Now I'm going to write something similar for attention, and then figure out how to go back to the format that RBERTviz needs for that (and submit a PR for RBERTviz at the same time to do that).

This currently adds a dplyr requirement. I think technically it could be rewritten in base (with some struggling for groups), but it'll be rough. For now I'm pretty sure the overhead needed for dplyr is far smaller than the overhead for BERT, so I don't think it's an issue.

File: R/run_classifier.R [hunk 1]
```diff
@@ -826,11 +810,11 @@ convert_examples_to_features <- function(examples,
     examples,
     function(ex_index, example,
              label_list, max_seq_length, tokenizer) {
-      if (ex_index %% 10000 == 1) {
-        # Print out every 10K starting with the first
-        print(paste("Converting example", ex_index,
-                    "of", num_examples ))
-      }
+      # if (ex_index %% 10000 == 1) {
```
Comment (jonathanbratt) [inline R/run_classifier.R]: should we just delete this?

File: R/extract_features.R [hunk 2]
```diff
@@ -188,11 +183,12 @@ input_fn_builder_EF <- function(features,
...
 #' @return \code{model_fn} closure for \code{TPUEstimator}.
 #' @keywords internal
-model_fn_builder_EF <- function(bert_config,
+.model_fn_builder_EF <- function(bert_config,
                              init_checkpoint,
                              layer_indexes,
                              use_tpu,
-                             use_one_hot_embeddings) {
+                             use_one_hot_embeddings,
+                             include_zeroth) {
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply R/extract_features.R]: Oops! Leftover from a failed attempt to make things easier. Deleted.

File: R/extract_features.R [hunk 3]
```diff
@@ -564,71 +554,96 @@ extract_features <- function(examples, # list of InputExamples_EF
...
-                                          seq_len(num_tokens),
-                                          seq_len(num_tokens)]
-      # Just return matrix as-is for now.
-      this_seq_attn[[key_str]] <- layer_attention
+  if (wants_output) {
+    if (wants_attention) {
+      return(list("layer_outputs" = big_output,
+                  "attention_probs" = big_attention))
+    } else {
+      return(big_output)
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply R/extract_features.R]: I thought about it, but when you just want one, it'd be a pain to get it into the format you want.

[the remark this answers is not in the document]
Comment (jonathanbratt) [reply R/extract_features.R]: Not *that* much of a pain. Seems like the potential pain from inconsistent output formats would be worse. Suppose I have some process where I always want to get the output embeddings, but only sometimes include the attention weights. I'd have to check for that when I handle the output piece.

**Pull request merged**